opensearch/catalog-opensearch-source/src/main/java/ddf/catalog/source/opensearch/OpenSearchSource.java

```diff
@@ -595,12 +622,49 @@ public String getVersion() {
         return "2.0";
     }
 
-    public void setInputTransformer(InputTransformer inputTransformer) {
-        this.inputTransformer = inputTransformer;
+    private InputTransformer getInputTransformer(InputStream inputStream) {
+        XMLStreamReader xmlStreamReader = null;
+        try {
+            xmlStreamReader = xmlInputFactory.createXMLStreamReader(inputStream);
+            Bundle bundle = FrameworkUtil.getBundle(this.getClass());
+            if (bundle == null) {
+                // only happens during unit tests
+                return defaultInputTransformer;
+            }
+            BundleContext bundleContext = bundle.getBundleContext();
```

Comment:
According to the docs, it can only be null when the bundle is not in the STARTING, STOPPING, or ACTIVE states, in which case this code is inaccessible.

https://osgi.org/javadoc/r4v43/core/org/osgi/framework/Bundle.html#getBundleContext%28%29